frontend/src/metabase/entities/questions.js

```diff
@@ -34,12 +34,24 @@ const Questions = createEntity({
         ),
       ),
 
-    setCollection: ({ id, model }, collection, opts) =>
-      Questions.actions.update(
-        { id },
-        { collection_id: canonicalCollectionId(collection && collection.id) },
-        undo(opts, model === "dataset" ? "model" : "question", "moved"),
-      ),
+    setCollection: ({ id, model }, collection, opts) => {
+      return async dispatch => {
+        const result = await dispatch(
+          Questions.actions.update(
+            { id },
+            {
+              collection_id: canonicalCollectionId(collection && collection.id),
+            },
+            undo(opts, model === "dataset" ? "model" : "question", "moved"),
+          ),
+        );
+        dispatch(
+          Collections.actions.fetchList({ tree: true }, { reload: true }),
```

Comment:
Without `reload: true` it won't reload in this case and will return cached data instead. https://github.com/metabase/metabase/blob/ac9bbc5b4a57ba3cba09fe739f120bd2d666f366/frontend/src/metabase/lib/redux.js#L301